api/client/src/main/java/org/projectnessie/client/api/ns/ClientSideGetNamespace.java

```diff
@@ -58,13 +58,13 @@ public GetNamespaceResult getWithResponse()
       if (!(c instanceof Namespace)) {
         throw new NessieNamespaceNotFoundException(
             contentKeyErrorDetails(key),
-            String.format("Namespace '%s' does not exist", key.toPathString()));
+            String.format("Namespace '%s' does not exist", key.toString()));
```

Comment:
Given the complexity of what we are trying to achieve, I wonder if relying on `toString()` is a good move, even for a mere error message.

Reply:
I don't have a good idea what to put into those messages.
`toPathString()` gives the control-character (which is often not really visible)
The encoded form looks weird to humans.
That's why I went with the native `.join(".")`

Reply:
I'd say we ought to define our own canonical `ContentKey` textual representation (consistent with servlet path rules). It would be useful for CLI parameters too.

As for `.toString()` in error messages, I personally prefer the less visually pleasing but unambiguous encoded form.

Reply:
`toHumanReadableString()` ?

Reply:
I think we have two major case:
1. Escaping the `.` separator - usable eveywhere full unicode is supported (e.g. CLI, JSON) - Let's call it `toCanonicalString()`?
2. Path escaping - the `.` plus all "suspicious" chars according to the new servlet spec - old `toPathString()` method.

I'd use (1) for error messages and `.toString()`. WDYT?

Reply:
+1 for `toCanonicalString()`

Reply:
@snazy : WDYT about `toCanonicalString()`?